### DISCUSSION TITLE
Getting error: IndexError: tuple index out of range

### DISCUSSION THREAD

**Discussion by steampunkchris** [thygate/stable-diffusion-webui-depthmap-script#87] [Q&A]
Sorry if this is the wrong place to ask.  I keep getting this message:
 
  File "C:\Users\***\AppData\Local\Programs\Python\Python310\lib\urllib\request.py", line 1351, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 11001] getaddrinfo failed>

Traceback (most recent call last):
  File "F:\AI\Super SD 2.0\stable-diffusion-webui\venv\lib\site-packages\gradio\routes.py", line 321, in run_predict
    output = await app.blocks.process_api(
  File "F:\AI\Super SD 2.0\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1016, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "F:\AI\Super SD 2.0\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 945, in postprocess_data
    if predictions[i] is components._Keywords.FINISHED_ITERATING:
IndexError: tuple index out of range

Is it perhaps an issue with my python?  sorry but I hope it is just me being lame.... TIA =)

**Comment by thygate** [comment]
When do you get this error ? 


There canadian sfu university server that hosts the pix2pix model is currently down, I wonder if it's related ... 

Do you get this error when trying to run depthmap with boost ? on a new install .. ?

**Reply by steampunkchris** [reply]
I think you are totally correct... The error is only shows up with BOOST on.  Hopefully the Canadian SFU server will be back up soon =)

Also, the midas_v21 has the same error with BOOST off.
but midas_v21_small works.  Actually they all work but midas_v21

Thank you again for the help! <3

**Reply by thygate** [reply]
Is there more error output in the console above those you posted already ?

The file it tries to download (/models/pix2pix/latest_net_G.pth) from the server that is down is only used when boost is enabled ..
edit:
Server seems to be back online atm, seems to be intermittent, there have been more reports ..
(https://sfu.ca/~yagiz/CVPR21/latest_net_G.pth)

**Reply by sordidloam** [reply]
@thygate Having a similar issue.  This mostly occurs when attempting to generate the "4 videos" although just now in testing was able to get the error without the 4 demo videos being selected.  

Update:  I did not have 'Generate 3D inpainted mesh' selected.  I'm trying that now.

I've tried res101, midasv21, midasv21small and different images, both simple and complex with similar results

NOTE: this has worked most of the time and is a WONDERFUL tool.  Thank you!

I get this message after a long string of text (I can provide upon request).

"Traceback (most recent call last):
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\routes.py", line 337, in run_predict
    output = await app.get_blocks().process_api(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1018, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 935, in postprocess_data
    if predictions[i] is components._Keywords.FINISHED_ITERATING:
IndexError: tuple index out of range"

**Comment by AugmentedRealityCat** [comment]
I had similar _Tuple_ errors yesterday at times. I'll document if it happens again.

**Comment by dangerweenie** [comment]
same error here (when trying to generated inpainted mesh):

`Inpaint 1:  43%|███████████████████████████████▎                                         | 3/7 [04:43<06:17, 94.43s/it]
Traceback (most recent call last):
  File "S:\GitHub\stable-diffusion-webui\venv\lib\site-packages\gradio\routes.py", line 337, in run_predict
    output = await app.get_blocks().process_api(
  File "S:\GitHub\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1018, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "S:\GitHub\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 935, in postprocess_data
    if predictions[i] is components._Keywords.FINISHED_ITERATING:
IndexError: tuple index out of range`

**Comment by sordidloam** [comment]
@thygate Same issue for me, ALTHOUGH for the first time it worked today, on a few lower resolution images.  I launched Stable Diffusion and updated to your latest release as of (3/18/23).  But then I went into do a few higher resolution images from my iphone, some of which were .JPEG, .png and .jpg.  My output is below...

DepthMap v0.3.8
device: cuda
Loading model weights from  ./models/leres/res101.pth
initialize network with normal
loading the model from ./models/pix2pix\latest_net_G.pth
Computing depthmap(s) ..
  0%|                                                                                            | 0/1 [00:00<?, ?it/s]

wholeImage being processed in : 1568
Adjust factor is: 1.0
Selecting patches ...
Target resolution:  (2352, 3136, 3)
Resulting depthmap resolution will be : (3024, 4032)
patches to process: 9
         processing patch 0 / 8 | [2304  144 1728 1728]
         processing patch 1 / 8 | [2304  576 1728 1728]
         processing patch 2 / 8 | [2304 1008 1728 1728]
         processing patch 3 / 8 | [ 720 1584 1440 1440]
         processing patch 4 / 8 | [1152 1584 1440 1440]
         processing patch 5 / 8 | [1584 1584 1440 1440]
         processing patch 6 / 8 | [2016 1584 1440 1440]
         processing patch 7 / 8 | [2448 1584 1440 1440]
         processing patch 8 / 8 | [   0 1728 1152 1152]
  0%|                                                                                            | 0/1 [00:12<?, ?it/s]
Error completing request
Arguments: (0, <PIL.Image.Image image mode=RGB size=4032x3024 at 0x2394761EE90>, None, '', '', 0, 0, 512, 512, False, False, True, True, True, False, False, 1, True, True, False, 2.5, 4, 0, False, 0, 1, True, True, 'u2net', False, False, False, 0, 2) {}
Traceback (most recent call last):
  File "E:\stable-diffusion-webui\modules\call_queue.py", line 56, in f
    res = list(func(*args, **kwargs))
  File "E:\stable-diffusion-webui\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "E:\stable-diffusion-webui\extensions\stable-diffusion-webui-depthmap-script\scripts\depthmap.py", line 1158, in run_generate
    outputs, mesh_fi = run_depthmap(None, outpath, imageArr, imageNameArr, compute_device, model_type, net_width, net_height, match_size, invert_depth, boost, save_depth, show_depth, show_heat, combine_output, combine_output_axis, gen_stereo, gen_stereotb, gen_anaglyph, stereo_divergence, stereo_fill, stereo_balance, clipdepth, clipthreshold_far, clipthreshold_near, inpaint, inpaint_vids, fnExt, vid_ssaa, background_removal, background_removed_images, save_background_removal_masks, False)
  File "E:\stable-diffusion-webui\extensions\stable-diffusion-webui-depthmap-script\scripts\depthmap.py", line 448, in run_depthmap
    images.save_image(Image.fromarray(img_output), path=outpath, basename=basename, seed=None, prompt=None, extension=opts.samples_format, info=info, short_filename=True,no_prompt=True, grid=False, pnginfo_section_name="extras", existing_info=None, forced_filename=None)
  File "E:\stable-diffusion-webui\modules\images.py", line 585, in save_image
    image = image.resize((round(opts.target_side_length), round(image.height * opts.target_side_length / image.width)), LANCZOS)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\PIL\Image.py", line 2192, in resize
    return self._new(self.im.resize(size, resample, box))
ValueError: image has wrong mode

Traceback (most recent call last):
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\routes.py", line 337, in run_predict
    output = await app.get_blocks().process_api(
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1018, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "E:\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 935, in postprocess_data
    if predictions[i] is components._Keywords.FINISHED_ITERATING:
IndexError: tuple index out of range

**Reply by sordidloam** [reply]
Also, it seems that now that this error has occurred, I can't get it to work at all, as if something was corrupted?

**Reply by sordidloam** [reply]
EDIT:  This ended up stalling, no error, just never finished and would not show progress update in the console.  

If I select 'Combine Into one image', the error goes away and I get to the go make some coffee statement.  So now it's cranking. 

Maybe that is required?  

![83f3d2d4906f744d79cb13fa178f6aab](https://user-images.githubusercontent.com/127781215/226094734-ae3b3276-baac-4d28-8bac-30e02c9d8207.png)

**Reply by sordidloam** [reply]
Me again.  Found a fix (I think).  I'm on a GTX 2080 Super with 8 GB.  The images were shot on an iPhone at full resolution, some 4k x 5k-ish in size at 72 dpi. I reduced the pixel count by half, 2k x 3k-ish and this worked.  So there appears to be an issue with higher res images. Maybe a warning could be built in or maybe this is only for those folks with 8 GB vram?  Not sure. (Didn't have the patience to test on CPU, much slower.)

Love this btw.  Amazing.
